src/Microsoft.AspNetCore.ResponseCaching/IResponseCache.cs

```diff
@@ -1,13 +1,14 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+
 namespace Microsoft.AspNetCore.ResponseCaching
 {
     public interface IResponseCache
     {
         object Get(string key);
-        // TODO: Set expiry policy in the underlying cache?
-        void Set(string key, object entry);
+        void Set(string key, object entry, TimeSpan validFor);
```

Comment:
What do we need in the context object? Should we add that when we actually need it?